mlflow/store/tracking/file_store.py

```diff
@@ -1356,6 +1357,9 @@ def start_trace(
         traces_dir = os.path.join(experiment_dir, FileStore.TRACES_FOLDER_NAME)
         mkdir(traces_dir, request_id)
         trace_dir = os.path.join(traces_dir, request_id)
+        # Q: do we need ARTIFACTS_FOLDER_NAME?
+        artifact_uri = os.path.join(trace_dir, FileStore.ARTIFACTS_FOLDER_NAME)
+        tags.update({MLFLOW_ARTIFACT_LOCATION: artifact_uri})
```

Comment:
```
- <experiment_id>
  - traces
    - <request_id>
      - ...
      - artifacts
        - traces.json
```

vs.

```
- <experiment_id>
  - traces
    - <request_id>
      - traces.json
```

Reply:
Not really sure if this use case really exists, but if we want to associate artifacts with a trace, the first option

Reply:
```
- <experiment_id>
  - traces
    - <request_id>
      - ...
      - traces.json
      - artifacts
        - ...
```


`traces.json` is a special file. Seems ok to save in the trace root directory. We can add an `artifacts` subdirectory in the future if necessary.

Reply:
Discussion offline: go with `artifacts/traces.json` path, make `traces.json` a reserved file if in the future we want to log more artifacts into traces